### PR TITLE
fix: command create cannot specify the namespace

### PR DESCRIPTION
**Pull request by zhzhuang-zju**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
The `karmadactl create` command currently does not support specifying the namespace for resources.

root cause：
Flags such as the `namespace` in `kubectl` are initialized at the kubectl level, and commands like `create` inherit these settings from there.
Since `karmadactl` itself does not initialize the `namespace` flag, the namespace flag needs to be initialized when directly using `cmd := kubectlcreate.NewCmdXXXX(f, ioStreams).`

solution：
solution 1: Initialize the `namespace` flag at the karmadactl level
Not feasible, because karmadactl commands such as `init` and `register`, which are specific to karmadactl, have special customizations for flags like `namespace`. Initializing the `namespace` at the karmadactl level would lead to conflicts.

solution 2：Each command initializes the `namespace` flag independently as needed, which is the approach adopted in this PR.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Might need to cherry-pick the changes back to the release-1.11 branch.

**Does this PR introduce a user-facing change?**:

```release-note
`karmadactl`：Fix the issue where commands `create`, `annotate`, `delete`,`edit`, `label` and `patch` cannot specify the namespace
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
Could you please give an example about how to reproduce it?

**Comment by zhzhuang-zju** [comment]
> Could you please give an example about how to reproduce it?

sure, when I'd like to update the labels on deployment `nginx` in the namespace `test`
```shell
$ karmadactl label deployment nginx resourcetemplate.karmada.io/deletion-protected=Always -n test
error: unknown shorthand flag: 'n' in -n
See 'karmadactl label --help' for usage.
$ karmadactl label deployment nginx resourcetemplate.karmada.io/deletion-protected=Always --namespace test
error: unknown flag: --namespace
See 'karmadactl label --help' for usage.
```

File: pkg/karmadactl/options/global.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ var DefaultConfigFlags = genericclioptions.NewConfigFlags(true).WithDeprecatedPa
 func AddKubeConfigFlags(flags *pflag.FlagSet) {
 	flags.StringVar(DefaultConfigFlags.KubeConfig, "kubeconfig", *DefaultConfigFlags.KubeConfig, "Path to the kubeconfig file to use for CLI requests.")
 	flags.StringVar(DefaultConfigFlags.Context, "karmada-context", *DefaultConfigFlags.Context, "The name of the kubeconfig context to use")
+	flags.StringVarP(DefaultConfigFlags.Namespace, "namespace", "n", *DefaultConfigFlags.Namespace, "If present, the namespace scope for this CLI request")
```
Comment (RainbowMango) [inline pkg/karmadactl/options/global.go:50]: I‘m afraid it's not appropriate to group `namespace` to kubeconfig flags.

File: pkg/karmadactl/options/global.go [hunk 2]
```diff
@@ -49,6 +49,11 @@ func AddKubeConfigFlags(flags *pflag.FlagSet) {
 	flags.StringVar(DefaultConfigFlags.Context, "karmada-context", *DefaultConfigFlags.Context, "The name of the kubeconfig context to use")
 }
 
+// AddNamespaceFlag add namespace flag to the specified FlagSet.
+func AddNamespaceFlag(flags *pflag.FlagSet) {
+	flags.StringVarP(DefaultConfigFlags.Namespace, "namespace", "n", *DefaultConfigFlags.Namespace, "If present, the namespace scope for this CLI request")
```
Comment (RainbowMango) [inline pkg/karmadactl/options/global.go:54]:
```suggestion
	flags.StringVarP(DefaultConfigFlags.Namespace, "namespace", "n", *DefaultConfigFlags.Namespace, "If present, the namespace scope for this CLI request.")
```
No big deal, but it feels better with a `.` at the end.

**Review by RainbowMango — COMMENTED** [review]
Looks much better now.
Given that this PR is supposed to be backported, we should ensure that it is fully tested. Please help provide the test result.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

/hold 
Please feel free to unhold this once the test result is provided.

**Comment by zhzhuang-zju** [comment]
> Given that this PR is supposed to be backported, we should ensure that it is fully tested. Please help provide the test result.

sure, here is my local verification

```shell
(base) ➜  karmada git:(master) ✗ karmadactl create configmap my-config --from-file=cert.pem -ntest
configmap/my-config created
(base) ➜  karmada git:(master) ✗ karmadactl get configmap -ntest 
NAME        CLUSTER   DATA   AGE   ADOPTION
my-config   Karmada   1      14s   -
(base) ➜  karmada git:(master) ✗ karmadactl delete cm my-config -ntest
configmap "my-config" deleted
(base) ➜  karmada git:(master) ✗ karmadactl get configmap -ntest      
No resources found in test namespace.
(base) ➜  karmada git:(master) ✗ karmadactl label deployment nginx resourcetemplate.karmada.io/deletion-protected=Exists -n test --overwrite
deployment.apps/nginx labeled
(base) ➜  karmada git:(master) ✗ karmadactl label deployment nginx resourcetemplate.karmada.io/deletion-protected- -n test 
deployment.apps/nginx unlabeled
(base) ➜  karmada git:(master) ✗ karmadactl annotate deployment nginx -ntest description='my frontend running nginx'
deployment.apps/nginx annotated
(base) ➜  karmada git:(master) ✗ karmadactl annotate deployment nginx -ntest description-                           
deployment.apps/nginx annotated
(base) ➜  karmada git:(master) ✗ karmadactl edit deployment nginx -ntest -oyaml
Edit cancelled, no changes made.
(base) ➜  karmada git:(master) ✗ karmadactl patch pp nginx-propagation --type='json' -p='[{"op": "replace", "path": "/spec/conflictResolution", "value":"Overwrite"}]' -ntest
propagationpolicy.policy.karmada.io/nginx-propagation patched
```

**Pull request merged**